src/events.rs

```diff
@@ -338,6 +346,229 @@ impl EventMask {
     }
 }
 
+/// A struct that provides structured access to event masks
+/// returned from reading an event from an inotify fd
+#[derive(Debug)]
+pub struct ParsedEventMask {
+    /// The kind of event that occurred
+    pub kind: Option<EventKind>,
+    /// The auxillary flags about the event
+    pub auxillary_flags: EventAuxillaryFlags,
+}
+
+impl ParsedEventMask {
+    /// Construct a `ParsedEventMask` from its component parts
+    pub fn from_parts(kind: Option<EventKind>, auxillary_flags: EventAuxillaryFlags) -> Self {
+        ParsedEventMask {
+            kind,
+            auxillary_flags,
+        }
+    }
+
+    /// Parse a raw event mask
+    pub fn from_raw_event_mask(mask: EventMask) -> Result<Self, EventMaskParseError> {
+        if mask.contains(EventMask::Q_OVERFLOW) {
+            return Err(EventMaskParseError::QueueOverflow);
+        }
+
+        let kind = mask.try_into()?;
+        let auxillary_flags = mask.into();
```

Comment:
_Very_ minor nitpick: These `into` and `try_into` calls can sometimes make code a bit hard to follow, since it's not always immediately obvious what type the value is being converted into. In general, unless when using an `impl Into<...>` parameter, I prefer to make it explicit using something like `EventAuxiliaryFlag::from(mask)`.